custom/icds_reports/reports/ls_launched.py

```diff
@@ -0,0 +1,310 @@
+from collections import OrderedDict, defaultdict
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import rrule, MONTHLY
+from django.db.models.aggregates import Sum, Max
+from django.utils.translation import ugettext as _
+
+from custom.icds_reports.cache import icds_quickcache
+from custom.icds_reports.const import LocationTypes, ChartColors, MapColors
+from custom.icds_reports.messages import ls_launched_help_text
+from custom.icds_reports.models.views import SystemUsageReportView
+from custom.icds_reports.utils import apply_exclude, indian_formatted_number, get_child_locations
+
+
+def get_prop(level):
+    if level == 1:
+        return 'states'
+    elif level == 2:
+        return 'districts'
+    elif level == 3:
+        return 'blocks'
+    elif level == 4:
+        return 'sectors'
+    else:
+        return 'awcs'
+
+
+@icds_quickcache(['domain', 'config', 'loc_level', 'show_test'], timeout=30 * 60)
+def get_ls_launched_data_map(domain, config, loc_level, show_test=False):
+    level = config['aggregation_level']
+
+    def get_data_for(filters):
+        filters['month'] = datetime(*filters['month'])
+        queryset = SystemUsageReportView.objects.filter(
+            **filters
+        ).values(
+            '%s_name' % loc_level, '%s_map_location_name' % loc_level
+        ).annotate(
+            states=Sum('num_launched_states') if level <= 1 else Max('num_launched_states'),
+            districts=Sum('num_launched_districts') if level <= 2 else Max('num_launched_districts'),
+            blocks=Sum('num_launched_blocks') if level <= 3 else Max('num_launched_blocks'),
+            sectors=Sum('num_launched_supervisors') if level <= 4 else Max('num_launched_supervisors'),
+            awcs=Sum('num_launched_awcs') if level <= 5 else Max('num_launched_awcs'),
```

Comment:
This wont be queried at awc level 5 yeah?

Reply:
level is 5, when a sector is selected